packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts

```diff
@@ -61,12 +62,14 @@ export interface IdDecodingContext {
 export function decode(
 	chunk: EncodedFieldBatch,
 	idDecodingContext: { idCompressor: IIdCompressor; originatorId: SessionId },
+	getIncrementalFieldBatch?: (fieldKey: string) => EncodedFieldBatch,
 ): TreeChunk[] {
 	return genericDecode(
 		decoderLibrary,
 		new DecoderContext(chunk.identifiers, chunk.shapes, idDecodingContext),
 		chunk,
 		anyDecoder,
+		getIncrementalFieldBatch,
```

Comment:
It seems like all the decoders need access to this "getIncrementalFieldBatch". Data which is needed during decoding and does not depend on exactly where you are in the tree should be put into the `TCache` (DecoderContext in this case), which is a generic type parameter so that different decoders can store what ever they need in it in a type safe way. This pattern prevents every time new data is needed from bloating the signatures from all the decode methods and adding lots of plumbing.